core/src/main/java/org/apache/accumulo/core/client/impl/TabletLocatorImpl.java

```diff
@@ -62,31 +61,15 @@
   // putting null, put MAX_TEXT
   static final Text MAX_TEXT = new Text();
 
-  private static class EndRowComparator implements Comparator<Text>, Serializable {
-
-    private static final long serialVersionUID = 1L;
-
-    @Override
-    public int compare(Text o1, Text o2) {
-
-      int ret;
-
-      if (o1 == MAX_TEXT)
-        if (o2 == MAX_TEXT)
-          ret = 0;
-        else
-          ret = 1;
-      else if (o2 == MAX_TEXT)
-        ret = -1;
-      else
-        ret = o1.compareTo(o2);
-
-      return ret;
-    }
-
-  }
-
-  static final EndRowComparator endRowComparator = new EndRowComparator();
+  static final Comparator<Text> endRowComparator = (o1, o2) -> {
+    if (o1 == o2)
+      return 0;
+    if (o1 == MAX_TEXT)
```

Comment:
`MAX_TEXT` really needs a better comment above it to better explain its role in this comparison (and also, so it doesn't get misused as a more general purpose "empty Text object").